gitlab/client.py

```diff
@@ -266,7 +266,8 @@ def lint(self, content: str, **kwargs: Any) -> Tuple[bool, List[str]]:
         """
         post_data = {"content": content}
         data = self.http_post("/ci/lint", post_data=post_data, **kwargs)
-        assert isinstance(data, dict)
+        if TYPE_CHECKING:
+            assert not isinstance(data, requests.Response)
```

Comment:
out of curiosity: we didn't actually see this anywhere right, all the http_* methods process the result in some way?

Reply:
> 
> 
> out of curiosity: we didn't actually see this anywhere right, all the http_* methods process the result in some way?

There are some calls where they do want the `requests.Response` object back from the http_* methods.  As far as I can tell things want either one or the other. I don't see any (yet) that might expect either back.